src/EnergyPlus/CurveManager.hh

```diff
@@ -453,36 +452,29 @@ namespace Curve {
 
 struct CurveManagerData : BaseGlobalStruct
 {
-    int NumCurves = 0;
-    bool GetCurvesInputFlag = true;
     bool CurveValueMyBeginTimeStepFlag = false;
     bool FrictionFactorErrorHasOccurred = false;
     bool showFallbackMessage = true;
-    EPVector<Curve::Curve *> PerfCurve;
+    Array1D<Curve::Curve *> curves;
+    std::map<std::string, int> curveMap;
```

Comment:
For fast lookup of curve of objects. The current pattern of using fast lookup (a map) to detect duplicate names but slow lookup (linear search) to find object indices is silly. If you already have a map, why not just use it for both? You can see this new pattern in `Schedule`, `Material`, and several other recently refactored modules also.